services/slam/builtin/builtin_test.go

```diff
@@ -1062,6 +1062,20 @@ func TestEndpointFailures(t *testing.T) {
 	test.That(t, fmt.Sprint(err), test.ShouldContainSubstring, "error getting the internal state from the SLAM client")
 	test.That(t, internalState, test.ShouldBeNil)
 
+	callbackPointCloud, err := svc.GetPointCloudMapStream(context.Background(), "hi")
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, callbackPointCloud, test.ShouldNotBeNil)
+	chunkPCD, err := callbackPointCloud()
+	test.That(t, err.Error(), test.ShouldContainSubstring, "Unimplemented desc")
+	test.That(t, chunkPCD, test.ShouldBeNil)
+
+	callbackInternalState, err := svc.GetInternalStateStream(context.Background(), "hi")
```

Comment:
Note: during testing, we sometimes make calls to GetInternalStateStream and sometimes to GetInternalStateFull. All errors are passed through in GetInternalStateFull which acts as a wrapper therefore it was deemed more space efficient to call GetInternalStateFull instead of GetInternalStateStream and aggregate the data, in all but certain units tests or tests of pure grpc streaming functionality